core/src/main/scala/org/http4s/blaze/pipeline/stages/SSLStage.scala

```diff
@@ -54,10 +54,30 @@ private object SSLStage {
   private case class SSLFailure(t: Throwable) extends SSLResult
 }
 
-final class SSLStage(engine: SSLEngine, maxWrite: Int = 1024 * 1024)
-    extends MidStage[ByteBuffer, ByteBuffer] {
+/** Default values for the [[SSLStage]] constructors.
+  *
+  * A separate `object` because the [[SSLStage]] `object` is `private`.
+  */
+object SSLStageDefaults {
+
+  /** Default value for [[SSLStage.maxWrite]] */
+  final val MaxWrite = 1024 * 1024
+}
+
+/** @param maxWrite
+  *   \@see [[SSLStageDefaults.MaxWrite]].
+  */
+final class SSLStage(
+    engine: SSLEngine,
+    maxWrite: Int,
+    sslHandshakeExceptionHandler: PartialFunction[Throwable, Unit]
```

Comment:
If it's expected to handle `SSLException`s only, shouldn't the contract be `PF[SSLException, Unit]`?